wurst/systems/Hints.wurst

```diff
@@ -0,0 +1,109 @@
+package Hints
+import ClosureTimers
+import LinkedList
+import ChatCommands
+import Game
+
+let TIME_BETWEEN_HINTS = 60.0
+let HINTS_OFF_MESSAGE = GENERAL_COLOR + "Your hints are now " + GOLD_COLOR + "off" + GENERAL_COLOR + "."
+let HINTS_ON_MESSAGE = GENERAL_COLOR + "Your hints are now " + GOLD_COLOR + "on" + GENERAL_COLOR + "."
+
+class Hint
+    
+    static force players = CreateForce()
+
+    private string message
+    private trigger action = null
+
+    construct(string hint)
+        this.message = hint
+
+    construct(string hint, code what)
+        this.message = hint
+        this.action = CreateTrigger()
+        TriggerAddAction(this.action, what)
+
+    static function pingUnit(unit which)
+        pingLoc(GetUnitLoc(which))
+
+    static function pingLoc(location loc)
+        PingMinimapLocForForceEx(players, loc, 5, bj_MINIMAPPINGSTYLE_FLASHY, 100, 100, 100)
+
+    static function addPLayer(player p)
+        ForceAddPlayer(players, p)
+        printTimedToPlayer(HINTS_ON_MESSAGE, 5, p)
+
+    static function removePlayer(player p)
+        ForceRemovePlayer(players, p)
+        printTimedToPlayer(HINTS_OFF_MESSAGE, 5, p)
+
+    static function togglePlayer(player p)
+        if (IsPlayerInForce(p, players))
+            Hint.removePlayer(p)
+        else    
+            Hint.addPLayer(p)
+
+    function display()
+        if IsPlayerInForce(GetLocalPlayer(), players)
+            printTimedToPlayer(GOLD_COLOR + "Hint: |r" + GENERAL_COLOR + message + "|r", 10, GetLocalPlayer())
+            if action != null
+                TriggerExecute(action)
+
+    static function initialize()
+        for i = 0 to 11
```

Comment:
This should be based off of a global I think.

Reply:
Can you elaborate on this? I don't fully understand what you mean/want me to do here.

Reply:
@Jaccouille I'll give you an example--

```
for i = 0 to bj_MAX_PLAYERS - 1
    if players[i].isIngame()
        ...
```